src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,54 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private int size;
+    private Object[] keys;
+    private Object[] values;
+
+    public StorageImpl() {
+        this.keys = new Object[MAX_SIZE];
+        this.values = new Object[MAX_SIZE];
+        this.size = 0;
+    }
+
+    private int getIndex(K key) {
+        for (int i = 0; i < size; i++) {
+            if (key == null && keys[i] == null) {
+                return i;
+            } else if (keys[i] != null && keys[i].equals(key)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+
+        if (index != -1) {
+            values[index] = value;
+        } else if (size < MAX_SIZE) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            throw new RuntimeException("Storage is full");
+        }
```

Comment:
Better to make majority of code without additional tabulations and minimize else blocks
With such approach code are simpler for reading and understanding
```suggestion
        if (index != -1) {
            values[index] = value;
            return;
        } 
        if (size >= MAX_SIZE) {
            throw new RuntimeException("Storage is full");
        } 
        keys[size] = key;
        values[size] = value;
        size++;
```